tests/epyccel/recognised_functions/test_numpy_funcs.py

```diff
@@ -3764,116 +3813,580 @@ def test_complex128():
-        arr = array([[4,5,6,2,1],[4,5,6,2,1]], int32)
+    @types('int32[:,:]')
+    def test_int32(arr):
+        from numpy import real, shape
         a = real(arr)
         s = shape(a)
         return len(s), s[0], s[1], a[0,1], a[1,0]
 
-    def test_int64():
-        from numpy import real, shape, array, int64
-        arr = array([[4,5,6,2,1],[4,5,6,2,1]], int64)
+    @types('int64[:,:]')
+    def test_int64(arr):
+        from numpy import real, shape
         a = real(arr)
         s = shape(a)
         return len(s), s[0], s[1], a[0,1], a[1,0]
 
-    def test_float():
-        from numpy import real, shape, array
-        arr = array([[4,5,6,2,1],[4,5,6,2,1]], float)
+    @types('float[:,:]')
+    def test_float(arr):
+        from numpy import real, shape
         a = real(arr)
         s = shape(a)
         return len(s), s[0], s[1], a[0,1], a[1,0]
 
-    def test_float32():
-        from numpy import real, shape, array, float32
-        arr = array([[4,5,6,2,1],[4,5,6,2,1]], float32)
+    @types('float32[:,:]')
+    def test_float32(arr):
+        from numpy import real, shape
         a = real(arr)
         s = shape(a)
         return len(s), s[0], s[1], a[0,1], a[1,0]
 
-    def test_float64():
-        from numpy import real, shape, array, float64
-        arr = array([[4,5,6,2,1],[4,5,6,2,1]], float64)
+    @types('float64[:,:]')
+    def test_float64(arr):
+        from numpy import real, shape
         a = real(arr)
         s = shape(a)
         return len(s), s[0], s[1], a[0,1], a[1,0]
 
-    def test_complex64():
-        from numpy import real, shape, array, complex64
-        arr = array([[4,5,6,2,1],[4,5,6,2,1]], complex64)
+    @types('complex64[:,:]')
+    def test_complex64(arr):
+        from numpy import real, shape
         a = real(arr)
         s = shape(a)
-        return len(s), s[0], s[1]
+        return len(s), s[0], s[1], a[0,1], a[1,0]
 
-    def test_complex128():
-        from numpy import real, shape, array, complex128
-        arr = array([[4,5,6,2,1],[4,5,6,2,1]], complex128)
+    @types('complex128[:,:]')
+    def test_complex128(arr):
+        from numpy import real, shape
         a = real(arr)
         s = shape(a)
-        return len(s), s[0], s[1]
+        return len(s), s[0], s[1], a[0,1], a[1,0]
+
+    import numpy as np
+    bl = randint(0, 1, size=(2, 5), dtype= bool)
+
+    integer = randint(min_int, max_int, size=(2, 5), dtype=np.int)
+    integer32 = randint(min_int32, max_int32, size=(2, 5), dtype=np.int32)
+    integer64 = randint(min_int64, max_int64, size=(2, 5), dtype=np.int64)
+
+    fl = uniform(min_float / 2, max_float / 2, size=(2, 5))
+    fl32 = uniform(min_float32, max_float32, size=(2, 5))
+    fl32 = np.float32(fl32)
+    fl64 = uniform(min_float64 / 2, max_float64 / 2, size=(2, 5))
+
+    cmplx64 = uniform(low=min_float32 / 2, high=max_float32 / 2, size=(2, 5)) + uniform(low=max_float32 / 2, high=max_float32 / 2, size=(2, 5)) * 1j
+    cmplx64 = np.complex64(cmplx64)
```

Comment:
I don't see the purpose of the second line.

Reply:
without the tests was failing with `cmplx64 should be 64 bits complex`

Reply:
by the way, can you please resolve what is already done.

Reply:
The first line is a 128 bit `complex` while the second line converts it to a 64 bit `complex64`. Maybe the first line should create a variable with a different name to make this clear?